libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -62,26 +66,43 @@ import kotlinx.coroutines.ExperimentalCoroutinesApi
  */
 @ExperimentalPreviewMapboxNavigationAPI
 @OptIn(ExperimentalCoroutinesApi::class)
-class NavigationView @JvmOverloads constructor(
+class NavigationView constructor(
     context: Context,
-    attrs: AttributeSet? = null,
-    accessToken: String = attrs.navigationViewAccessToken(context),
+    attrs: AttributeSet?,
+    accessToken: String,
+    hostingLifecycle: Lifecycle?,
+    viewModelStoreOwner: ViewModelStoreOwner?,
 ) : FrameLayout(context, attrs), LifecycleOwner {
 
+    @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        accessToken: String = attrs.navigationViewAccessToken(context),
+        hostingFragment: Fragment? = null
+    ): this(
```

Comment:
Handling defaults with @JvmOverloads and multiple constructors is a mess. Better to create the single default constructor and then builder functions